src/main/java/org/mitre/synthea/engine/State.java

```diff
@@ -438,9 +438,24 @@ public boolean process(Person person, long time) {
 
     public abstract long endOfDelay(long time, Person person);
 
+    /**
+     * Process any aspect of this state which should only happen once.
+     * Because of the nature of Delay states, this state may get called
+     * multiple times: once per timestep while delaying. To ensure
+     * any actions which are supposed to happen only happen once,
+     * this function should be overriden in subclasses.
+     *
+     * @param person the person being simulated
+     * @param time the date within the simulated world
+     */
+    public void processOnce(Person person, long time) {
```

Comment:
Shouldn't this be an `abstract` method as well, like `endOfDelay`?

Reply:
Yeah it could be. I saw 2 options -- (1)make this abstract and add an empty implementation in `Delay`, or (2)have an empty implementation here and `Delay` doesn't need any changes. I'm happy to change it if there's a strong preference.

Reply:
I was hoping to avoid an empty implementation, but I guess we can't do that... so we can leave it as is.